_docs/concepts/security/authn-policy.md

```diff
@@ -0,0 +1,100 @@
+
+{% include figure.html width='80%' ratio='100%'
+    img='./img/authn.png'
+    alt='Istio authentication policy architecture'
+    title='Istio authentication policy architecture'
+    caption='Istio authentication policy architecture'
+    %}
+
+
+
+Policy is scoped at namespace level, with (optional) target selector rules to narrow down the set of services (within the same namespace as the policy) on which the policy should be applied. This aligns with the ACL model based on Kubernetes RBAC. More specifically, only admin of the namespace can set policies for services in that namespace.
+
+
+Authentication engine is implemented on sidecar. For example, with Envoy sidecar, it is a combination of SSL settings and HTTP filters. If authentication fails, request will be rejected (either with SLL handshake error code, or http 401, depends on the type of authencation mechanism). If success, following authenticated attributes will be generated:
+
+- **source.principal**: peer principal. If peer authentiation is not used, the attribute is not set.
+- **request.auth.principal**: depends on the policy principal binding, this could be peer principal (if USE_PEER) or origin principal (if USE_ORIGIN).
+- **request.auth.aud**: reflect the audience (*aud*) claim within the origin-JWT (JWT that is used for origin authentication)
+- **request.auth.presenter**: similarly, reflect the authorize presenter (*azp*) claim.
+- **request.auth.claims**: all raw string claims from origin-JWT.
+
+Origin principal is not explicitely output. In general, it can always be reconstructed from issuer (*iss*) and subject (*sub*) claims. If principal binding is USE_ORIGIN, it is also the same as **request.auth.principal**.
+
+
+## Anatomy of the policy
+
+### Target selectors
+
+Defines rule to find service(s) on which policy should be applied. If no rule provided, the policy will be matched to all services in the namespace, so call namespace-level policy (as opposed to service-level policy are those that have non-empty selector rules). Istio (pilot) will pick the service-level policy if available, otherwise fallback to namespace-level policy. If none define, it uses the default policy based on service mesh config.
+
+
+Operators are responsible to avoid conflict, e.g create more than one service-level policy that match to the same service(s) (or more than one namespace-level policy on the same namespace).
+
+
+Example: rule to select product-page service (on any port), and reviews:9000.
+
+```
+ targets:
+ - name: product-page
+ - name: reviews
+   ports:
+   - number: 9000
+```
+
+### Peer authentication
+
+
+Defines authentication methods (and associated parameters) that are supported for for peer authentication. It can list more than one methods; only one of them need to be satisfied for the authentication pass. However, in the early releases, only mutual TLS is supported. Omitting this if peer authentication is not needed.
```

Comment:
for for -> for
one of them need -> one of them needs
the authentication pass -> the authentication to pass

Reply:
Done